app/Model/Attribute.php

```diff
@@ -3526,4 +3513,66 @@ private function generateTypeDefinitions()
             //'url-regex' => array('desc' => '', 'default_category' => 'Person', 'to_ids' => 0),
         );
     }
+
+    private function findAttributeByValue($value)
+    {
+        $type = $this->data['Attribute']['type'];
+        $conditions = [
+            'Attribute.event_id' => $this->data['Attribute']['event_id'],
+            'Attribute.type' => $type,
+            'Attribute.deleted' => 0,
+            'Attribute.object_id' => 0,
+        ];
+
+        if (isset($this->data['Attribute']['category'])) {
+            $conditions['Attribute.category'] = $this->data['Attribute']['category'];
+        }
+
+        if (in_array($type, $this->getCompositeTypes(), true)) {
+            $value = explode('|', $value);
+            $conditions['Attribute.value1'] = $value[0];
+            $conditions['Attribute.value2'] = $value[1];
+        } else {
+            $conditions['Attribute.value1'] = $value;
+        }
+
+        if (isset($this->data['Attribute']['id'])) {
+            $conditions['Attribute.id !='] = $this->data['Attribute']['id'];
+        }
+
+        return $this->find('first', [
+            'recursive' => -1,
+            'conditions' => $conditions,
+            'fields' => ['Attribute.id', 'Attribute.uuid']
+        ]);
+    }
+
+    /**
+     * If there is no id set but attribute value is set, check if the attribute already exists (by value) 
+     *
+     * {@inheritdoc }
+     *
+     * @param int|string $id ID of record to check for existence
+     * @return bool True if such a record exists
+     */
+    public function exists($id = null)
```

Comment:
Why is the override needed?

Reply:
`cakephp` uses this functions to check if it's a new or existing entity, this happens in `Model::_doSave()`, before `beforeSave()` is called.
The alternative would be add additional checks to `Attribute:captureAttribute()` but this won't cover all the other places `Attribute->save()` is called directly